astropy/coordinates/tests/test_solar_system.py

```diff
@@ -437,9 +437,13 @@ def test_file_ephemeris_wrong_input():
     # Try loading a non-existing file:
     with pytest.raises(ValueError):
         get_body('earth', time, ephemeris='/path/to/nonexisting/file.bsp')
+
+    # This test currently leaves the file open. To fix this issue, an
+    # upstream fix is required in jplephem package. This test is removed
+    # until that issue is fixed.
     # Try loading a file that does exist, but is not an ephemeris file:
-    with pytest.raises(ValueError):
-        get_body('earth', time, ephemeris=__file__)
+    # with pytest.raises(ValueError):
```

Comment:
Why not check for `jplephem` version and skip? Then it will naturally run again when new version is out?

Reply:
I haven't submitted a PR to `jplephem` yet and I have no idea what version of `jplephem` will fix this.

Reply:
How about we also stick in a `with pytest.warns(ResourceWarning)` here. At least then we know when the test starts failing with new version. I am afraid if it is just commented out that we'll forget about it.

Reply:
`gc.collect()` did nothing here.  I kept the test, but added the `with pytest.warns(ResourceWarning)`.

Indeed, it's very annoying!